pandas/core/missing.py

```diff
@@ -585,6 +589,9 @@ def _interpolate_scipy_wrapper(
         "cubic",
         "polynomial",
     ]
+    valid = NP_METHODS + SP_METHODS
+    if method not in valid:
+        raise ValueError(f"Can not interpolate with method={method}.")
```

Comment:
Instead of this, the `terp = alt_methods[method]` below should be changed to:

```python
terp = alt_methods.get(method, None):
if terp is None:
    raise ValueError(f"Can not interpolate with method={method}.")
```